docs/multi_process.md

```diff
@@ -109,17 +109,17 @@ function is global. The mesh is also global.
 TODO: xmap example
 
 **It’s very important that all processes run the same cross-process computations
-in the same order.** This usually happens naturally if you run the same JAX
-Python program in each process. Some common pitfalls to look out for that may
-cause differences:
-
-*   Different devices running differently-shaped shards of the data, which can
-    cause hangs or incorrect return values. This can be caused by different
-    processes passing in differently-shaped inputs to the same parallel
-    function. Differently-shaped inputs are only a problem if they result in
-    differently-shaped shards across processes; e.g. passing in different
-    leading batch sizes in order to run on different numbers of local devices
-    per process is ok.
+in the same order.** Running the same JAX Python program in each process is
+usually sufficient. Some common pitfalls to look out for that may cause
+differently-ordered computations despite running the same program:
+
+*   Processes passing differently-shaped inputs to the same parallel function
+    can cause hangs or incorrect return values. Differently-shaped inputs are
+    safe so long as they result in identically-shaped per-device data shards
+    across processes; e.g. passing in different leading batch sizes in order to
+    run on different numbers of local devices per process is ok, but having each
+    process pad its batch to a different max example length is not.
```

Comment:
I did my best with this part, but it's super tricky to explain clearly and concisely (at least for me). Any further suggestions particularly welcome here.

Reply:
This is fine for now, let's not stress about it too much.